vulnfeeds/cmd/nvd-cve-osv/main.go

```diff
@@ -168,29 +167,51 @@ func GitVersionsToCommits(CVE cves.CVEID, versions cves.VersionInfo, repos []str
 					Logger.Warnf("[%s]: Failed to get a Git commit for fixed version %q from %q: %v", CVE, av.Fixed, repo, err)
 				} else {
 					Logger.Infof("[%s]: Successfully derived %+v for fixed version %q", CVE, ac, av.Fixed)
-					if v.Duplicated(ac) {
-						Logger.Warnf("[%s]: Duplicate: %#v already present in %#v", CVE, ac, v)
-					} else {
-						v.AffectedCommits = append(v.AffectedCommits, ac)
-					}
+					fixedEquivalentCommit = ac.Fixed
 				}
 			}
 			// Only try and convert last_affected versions to commits via tags if there aren't any Fixed commits already (to maintain schema compliance).
 			// cves.ExtractVersionInfo() opportunistically returns
 			// AffectedCommits (with Fixed commits) when the CVE has appropriate references.
+			lastAffectedEquivalentCommit := ""
 			if !v.HasFixedCommits(repo) && av.LastAffected != "" {
 				ac, err := git.VersionToCommit(av.LastAffected, repo, cves.LastAffected, normalizedTags)
 				if err != nil {
 					Logger.Warnf("[%s]: Failed to get a Git commit for last_affected version %q from %q: %v", CVE, av.LastAffected, repo, err)
 				} else {
 					Logger.Infof("[%s]: Successfully derived %+v for last_affected version %q", CVE, ac, av.LastAffected)
-					if v.Duplicated(ac) {
-						Logger.Warnf("[%s]: Duplicate: %#v already present in %#v", CVE, ac, v)
-					} else {
-						v.AffectedCommits = append(v.AffectedCommits, ac)
-					}
+					lastAffectedEquivalentCommit = ac.LastAffected
 				}
 			}
+			// Assemble a single AffectedCommit from what was resolved, iff it
+			// doesn't result in a half-resolved (false positive-causing)
+			// situation with a successfully resolved introduced version and an
+			// unsuccessfully resolved fixed or last_affected version.
+			ac := cves.AffectedCommit{}
+			if av.Introduced != "" && introducedEquivalentCommit != "" && (av.Fixed != "" && fixedEquivalentCommit != "") || (av.LastAffected != "" && lastAffectedEquivalentCommit != "") {
```

Comment:
Why do we need to check `av.Introduced != ""` here? Would `introducedEquivalentCommit != ""` already imply that?

Reply:
Similarly for the fix, last affected counterparts also.

Reply:
Yes, you're right, thanks. Simplified in 5d3ce3dc